packages/component/src/ConnectivityStatus/Connecting.js

```diff
@@ -0,0 +1,63 @@
+import classNames from 'classnames';
+import PropTypes from 'prop-types';
+import React, { useState } from 'react';
+
+import ScreenReaderText from '../ScreenReaderText';
+import SpinnerAnimation from './Assets/SpinnerAnimation';
+import useForceRender from '../hooks/internal/useForceRender';
+import useLocalize from '../hooks/useLocalize';
+import useStyleOptions from '../hooks/useStyleOptions';
+import useStyleSet from '../hooks/useStyleSet';
+import useTimer from '../hooks/internal/useTimer';
+import WarningNotificationIcon from './Assets/WarningNotificationIcon';
+
+const ConnectivityStatusConnecting = ({ reconnect }) => {
+  const [{ slowConnectionAfter }] = useStyleOptions();
+  const [
+    { connectivityNotification: connectivityNotificationStyleSet, warningNotification: warningNotificationStyleSet }
+  ] = useStyleSet();
+  const [initialRenderAt] = useState(() => Date.now());
+  const connectivityStatusLabelText = useLocalize('ConnectivityStatus');
+  const initialConnectionText = useLocalize('INITIAL_CONNECTION_NOTIFICATION');
+  const interruptedConnectionText = useLocalize('INTERRUPTED_CONNECTION_NOTIFICATION');
+  const slowConnectionText = useLocalize('SLOW_CONNECTION_NOTIFICATION');
+  const forceRender = useForceRender();
+
+  useTimer(initialRenderAt + slowConnectionAfter, forceRender);
```

Comment:
Should we have a check to make sure `slowConnectionAfter` is non-negative? Also what happens if it's less than 400ms?

Reply:
Will check for non-negative in Composer.

Debounce is used for delaying updates/dismisses. If `slowConnectionAfter` is set to zero or negative, since there are no changes (initially slow), the notification should show up as slow connection initially.